### PR TITLE
Spring Security Integration

### PR DESCRIPTION
**Pull request by sronderos**
Would someone on the project team mind checking over these changes before they are commited to the main line?

I've tested the functionality as best I can, I'd like to know if anyone objects the the way that I've integrated Spring Secuirty.  There is probably a lot of room for improvement on this script, so another question is how much more does it need before it can be released?

### CODE REVIEW

File: src/templates/SpringSecurityTenantResolverSkeleton.groovy.template [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * Spring Security Tenant Resolver
+ * Extracts the userTenantId 
+ * @see http://multi-tenant.github.com/grails-multi-tenant-single-db/
+ */
+class ${tenantResolverClassName} implements TenantResolver {
+    SpringSecurityService springSecurityService
+    
+    @Override
+    Integer resolve(HttpServletRequest request) throws TenantResolveException {
+        def user = springSecurityService.getCurrentUser()
```
Comment (kimble) [inline src/templates/SpringSecurityTenantResolverSkeleton.groovy.template:19]: Would _user_ typically be an instance of a class implementing the _Tenant_ interface?

**Comment by kimble** [comment]
Looks great to me! 

I've started in a new job so I'm not that invested in this plugin anymore, but after spending quite a bit of time working on the plugin it makes me very happy to see that someone is still working on it!

File: src/templates/SpringSecurityTenantResolverSkeleton.groovy.template (hunk 1 above)
Comment (sronderos) [inline src/templates/SpringSecurityTenantResolverSkeleton.groovy.template:19]: No, I figured that a user has one tenant but one tenant could have many users.
'user' in this case would be the user domain from Spring Security's perspective.  I'll need to also add this part to the documentation, but how I planned for it to work is that the Spring Security user domain will need to have a userTenantId field added to it which matches the id of the Tenant implementation.

Comment (kimble) [inline src/templates/SpringSecurityTenantResolverSkeleton.groovy.template:19]: I see! Perhaps we should add an interface for classes related to a tenant? I know that it isn't very "groovy-ish", but personally I find that it makes code easier to read.

Comment (sronderos) [inline src/templates/SpringSecurityTenantResolverSkeleton.groovy.template:19]: I considered that as well, but if we go that far it seems like we start getting into an area where we should have a separate plugin.  I don't really know how to do the annotation that would integrate with Spring Security without relying on spring-security-core at compile time.

Comment (kimble) [inline src/templates/SpringSecurityTenantResolverSkeleton.groovy.template:19]: That's a very fair point. Adding spring-security-core as a compile time dependency is not an option. Anyway, I look at those scripts as more of a help-me-get-started kinda thing then ready production code. 

Feel free to merge changes whenever you feel like it. Oh, an old version has been published to grails.org by LucasTeixeira. You can probably get in touch with him for more information if you want to publish a new version to the grails plugin repository.

**Comment by lucastex** [comment]
Hi guys...

I'm already in contact with sronderos, we'll get this working.

[]s,

**Pull request merged**